modules/aws/rds.go

```diff
@@ -86,15 +86,91 @@ func GetParameterValueForParameterOfRdsInstance(t testing.TestingT, parameterNam
+	for morepages == true {
+		//  *** on the first pass, the lastmarker is nil, if additional records exist, the api call will return a marker. cnt holds the number of items in output
+		output, marker, cnt := GetParametersOfRdsInstance(t, dbInstanceID, awsRegion, lastmarker)
+		// store the new marker to a local var
+		lastmarker = marker
+		//  check if the number of records returned are less than 100 or if a marker was not returned to have function stop on this current page
+		if (cnt < 100) || (marker == "") {
+			morepages = false
 		}
+		
+		// examine the parameters returned for any matches to the specified
+		for _, parameter := range output {
+			//  ### debug print each of the parameters to match to the parameterName 
+			// fmt.Println(aws.StringValue(parameter.ParameterName))
+			if aws.StringValue(parameter.ParameterName) == parameterName {
+				return aws.StringValue(parameter.ParameterValue), nil
+			}
+		}
+		
+		//  the above code should prevent paging beyond the actual records but just in case, included a stop
+		pagecnt++
+		if pagecnt > 6 {
+			return "", fmt.Errorf("Error: scanned beyond 6 pages")
+		}
+	}
+	return "", aws.ParameterForDbInstanceNotFound{ParameterName: parameterName, DbInstanceID: dbInstanceID, AwsRegion: awsRegion}
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------------
+//------------------------------------------------------------------------------------------------------------------------------------------------
+// GetParametersOfRdsInstance gets all the parameters defined in the parameter group for the RDS instance in the given region.
+// Modified to accept a Marker Token to pass on to GetParametersOfRdsInstanceE and return the ending marker and the item count in parameters
+func GetParametersOfRdsInstance(t testing.TestingT, dbInstanceID string, awsRegion string, startMarker string) ([]*rds.Parameter, string, int) {
+	parameters, endmarker, cnt, err := GetParametersOfRdsInstanceE(t, dbInstanceID, awsRegion, startMarker)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return parameters, endmarker, cnt
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------------
+//------------------------------------------------------------------------------------------------------------------------------------------------
+// GetParametersOfRdsInstanceE gets all the parameters defined in the parameter group for the RDS instance in the given region.
+// Modified to accept a Marker Token to pass on to DescribeDBParameters API call. Returns rds parameters, the ending marker from search, and count of records pulled
+func GetParametersOfRdsInstanceE(t testing.TestingT, dbInstanceID string, awsRegion string, strtmarker string) ([]*rds.Parameter, string, int, error) {
```

Comment:
I don't think it is very ergonomical to replace and push the pagination details up from this function, especially when the expected test use case would be for RDS instances with very few parameters.

It probably makes sense to create two functions, one that returns all objects (the original function but modified to collect all the pages), and another function that returns partial results, which is this one.

That said, I would probably tweak the function API to use the paginator pattern, where a function is called on each page. The function signature would be something like:

```
func GetParametersOfRdsInstancePaginatorE(t testing.TestingT, dbInstanceID string, awsRegion string, func([]* rds.Parameter) error) error
```